parsing/dialects/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -2789,6 +2789,235 @@ SqlSelect SqlSelect() :
+                    throw SqlUtil.newContextException(getPos(), RESOURCE.applyNotAllowed());
+                }
+                e = new SqlJoin(joinType.getParserPosition(),
+                    e,
+                    SqlLiteral.createBoolean(false, joinType.getParserPosition()),
+                    joinType,
+                    e2,
+                    JoinConditionType.ON.symbol(SqlParserPos.ZERO),
+                    SqlLiteral.createBoolean(true, joinType.getParserPosition()));
+            }
+        )
+    )*
+    {
+        return e;
+    }
+}
+
+SqlNode NonqueryTable() :
+{
+    SqlNode tableRef;
+    final SqlNode over;
+    final SqlNode snapshot;
+    final SqlNode match;
+    SqlNodeList extendList = null;
+    SqlUnnestOperator unnestOp = SqlStdOperatorTable.UNNEST;
+    final Span s, s2;
+    SqlNodeList args;
+}
+{
+    (
+        LOOKAHEAD(2)
+        tableRef = TableRefWithHintsOpt()
+        [
+            [ <EXTEND> ]
+            extendList = ExtendList() {
+                tableRef = extend(tableRef, extendList);
+            }
+        ]
+        over = TableOverOpt() {
+            if (over != null) {
+                tableRef = SqlStdOperatorTable.OVER.createCall(
+                    getPos(), tableRef, over);
+            }
+        }
+        [
+            tableRef = Snapshot(tableRef)
+        ]
+        [
+            tableRef = MatchRecognize(tableRef)
+        ]
+    |
+        <UNNEST> { s = span(); }
+        args = ParenthesizedQueryOrCommaList(ExprContext.ACCEPT_SUB_QUERY)
+        [
+            <WITH> <ORDINALITY> {
+                unnestOp = SqlStdOperatorTable.UNNEST_WITH_ORDINALITY;
+            }
+        ]
+        {
+            tableRef = unnestOp.createCall(s.end(this), args.toArray());
+        }
+    |
+        <TABLE> { s = span(); } <LPAREN>
+        tableRef = TableFunctionCall(s.pos())
+        <RPAREN>
+    |
+        tableRef = ExtendedTableRef()
+    )
+    { return tableRef; }
+}
+
+SqlNode TableAlias(SqlNode tableRef) :
+{
+    final SqlIdentifier alias;
+    SqlNodeList columnAliasList = null;
+}
+{
+    [
+        [ <AS> ] alias = SimpleIdentifier()
+        [ columnAliasList = ParenthesizedSimpleIdentifierList() ]
+        {
+            if (columnAliasList == null) {
+                tableRef = SqlStdOperatorTable.AS.createCall(
+                    Span.of(tableRef).end(this), tableRef, alias);
+            } else {
+                List<SqlNode> idList = new ArrayList<SqlNode>();
+                idList.add(tableRef);
+                idList.add(alias);
+                idList.addAll(columnAliasList.getList());
+                tableRef = SqlStdOperatorTable.AS.createCall(
+                    Span.of(tableRef).end(this), idList);
+            }
+        }
+    ]
+    {
+        return tableRef;
+    }
+}
+
+/** Matches "LEFT JOIN t ON ...", "RIGHT JOIN t USING ...", "JOIN t". */
+SqlNode JoinClause(SqlNode e) :
+{
+    SqlNode e2, condition;
```

Comment:
nit: Since you only assign value to e2 once, it might be good to have it declared with final keyword.

Reply:
SGTM